src/nuc/token.py

```diff
@@ -0,0 +1,240 @@
+"""
+NUC tokens.
+"""
+
+import json
+import re
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import Any, Dict, List
+
+from .policy import Policy
+
+_DID_PATTERN = re.compile("did:([a-zA-Z0-9_-]+):([a-zA-Z0-9]{66})")
+_HEX_PATTERN = re.compile("[a-zA-Z0-9]+")
+
+
+@dataclass
+class Did:
+    """
+    A class representing a Decentralized Identifier (DID).
+    """
+
+    method: str
+    public_key: bytes
+
+    @staticmethod
+    def nil(public_key: bytes) -> "Did":
+        """
+        Construct a new DID for the "nil" method.
+        """
+
+        return Did("nil", public_key)
+
+    @staticmethod
+    def parse(data: str) -> "Did":
+        """
+        Parse a DID from a string.
+        """
+        matches = _DID_PATTERN.findall(data)
+        if not matches:
+            raise MalformedDidException("invalid DID")
+
+        (method, public_key) = matches[0]
+        try:
+            public_key = bytes.fromhex(public_key)
+        except Exception as ex:
+            raise MalformedDidException("invalid hex public key") from ex
+        return Did(method, public_key)
+
+    def __str__(self) -> str:
+        return f"did:{self.method}:{self.public_key.hex()}"
+
+
+@dataclass
+class Command:
+    """
+    A command to be invoked.
+    """
+
+    segments: List[str]
+
+    @staticmethod
+    def parse(data: str) -> "Command":
+        """
+        Parse a command from a string.
+        """
+
+        if not data.startswith("/"):
+            raise MalformedCommandException("commands must start with '/'")
```

Comment:
Do we allow `cmd: /` or are we enforcing either `cmd: /nil` " `cmd: /nuc`?

Reply:
I think this type of enforcement needs to happen at token validation time no?